### ISSUE TITLE
Functional test failing in OSX El Capitan

### ISSUE THREAD

**Issue by elmart** [neovim/neovim#3501]
ShaDa related funcional test failing in master:
```
859 successes / 1 failure / 0 errors / 5 pending : 5.928544 seconds

Pending → ...ojects/os/neovim/neovim/test/functional/job/job_spec.lua @ 362
jobs will only emit the "exit" event after "stdout" and "stderr"

Pending → .../neovim/neovim/test/functional/provider/python3_spec.lua @ 10
Python 3 (or the Python 3 neovim module) is broken or missing:
provider/pythonx: Could not load Python 3:
python3 not found in search path or not executable.
python3.5 not found in search path or not executable.
python3.4 not found in search path or not executable.
python3.3 not found in search path or not executable.
/opt/local/bin/python is Python 2.7 and cannot provide Python 3.

Pending → ...neovim/neovim/test/functional/shell/viml_system_spec.lua @ 189
system() with a program that doesn't close stdout skipped (missing xclip)

Pending → ...neovim/neovim/test/functional/shell/viml_system_spec.lua @ 368
systemlist() with a program that doesn't close stdout skipped (missing xclip)

Pending → ...s/neovim/neovim/test/functional/ui/screen_basic_spec.lua @ 555
Screen resize has minimum width/height values

Failure → ...ts/os/neovim/neovim/test/functional/shada/marks_spec.lua @ 103
ShaDa support code is able to populate v:oldfiles
...ts/os/neovim/neovim/test/functional/shada/marks_spec.lua:112: Expected objects to be the same.
Passed in:
(string) 'estfile-functional-shada-marks-2'
Expected:
(string) 'Xtestfile-functional-shada-marks'

stack traceback:
	...ts/os/neovim/neovim/test/functional/shada/marks_spec.lua:112: in function <...ts/os/neovim/neovim/test/functional/shada/marks_spec.lua:103>

-- Output to stderr:

CMake Error at /Users/eliseo/projects/os/neovim/neovim/cmake/RunTests.cmake:39 (message):
  Running functional tests failed with error: 1.
```
Also, built binary doesn't seem to work well (.vimrc apparently not read).

**Comment by ZyX-I** [comment]
`.vimrc` was not read for months. `.nvimrc` is no longer read since we support XDG base directory specification, check `:h vimrc`. Not sure why the test fails though.

**Comment by jszakmeister** [comment]
It's not just El Capitan that's affected.  The same test fails on master under Linux too.

**Comment by tarruda** [comment]
I also had this failure after rebasing #3413 on the ShaDa changes on my linux VM, but only when using the `~/dev/neovim` directory as the source root, which simply happens to be where I always kept my local neovim git clone.

I did not give it much importance since it is probably caused by test environment garbage(which is why travis is not affected), but here's what I found out:

- The problem is gone after moving ~/dev/neovim to another directory(~/neovim) and doing `make distclean test`
- The problem is there after cloning a fresh copy into `~/dev/neovim`

This leads me to conclude that whatever is affecting the test lives outside the neovim directory(Could it be a problem caused by viminfo -> Shada migration?)

**Comment by justinmk** [comment]
I saw the test fail randomly on Travis, seems to be a race.

**Comment by liujoey** [comment]
el capitan, master branch not loading .nvimrc and .nvim folder

**Comment by jszakmeister** [comment]
A little more information...  I printed out the value of `v:oldfiles` and it's not in the order the test expects it:

> ['/home/quickbuild/buildagent/workspace/root/neovim/pull-requests/Xtestfile-functional-shada-marks2', '/home/quickbuild/buildagent/workspace/root/neovim/pull-requests/Xtestfile-functional-shada-marks']

**Comment by mhinz** [comment]
I think this one is fixed?

**Comment by jszakmeister** [comment]
Yes, it was fixed by a6c45d15fd34892ad93a7cfcf678f05b8ed56ed3.  Closing.
